server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/LocalDelivery.java

```diff
@@ -119,7 +119,6 @@ public void init(MailetConfig mailetConfig) throws MessagingException {
             .resourceLocator(ResourceLocatorImpl.instanciate(usersRepository, sieveRepository))
             .consume(getInitParameter("consume", true))
             .deliveryHeader("Delivered-To")
-            .resetReturnPath(true)
```

Comment:
This job is also done by RRT.
It's normal to ensure return path matches the enveloppe. I think we don't need to expose it.